### PR TITLE
feat: assign config paths in build dependencies in cache config

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
feat

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
https://webpack.js.org/configuration/other-options/#cachebuilddependencies

- After resolving core flags, assigns the configs as a build dep which is invalidated when configs change

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: test/core-flags/cache-flags.test.js [hunk 1]
```diff
@@ -69,4 +68,37 @@ describe('cache related flags from core', () => {
...
+        expect(newRun.stdout).toContain('[cached] 1 module');
+
+        // Change config to invalidate cache
+        writeFileSync(resolve(__dirname, './webpack.test.config.js'), 'module.exports = {mode: "production"}');
+
+        const newRun2 = run(__dirname, ['--cache-type', 'filesystem', '-c', './webpack.test.config.js']);
+        unlinkSync(resolve(__dirname, './webpack.test.config.js'));
+        expect(newRun2).not.toContain('[cached] 1 module');
+        expect(newRun2.exitCode).toEqual(0);
+    });
```
Comment (evilebottnawi) [inline test/core-flags/cache-flags.test.js:103]: Can we add test where `cache: { type: ''filesystem }` in config and we inject `buildDependencies` too?

Also can we use here WebpackCLIPlugin for tests on output `cache` value

Comment (anshumanv) [reply test/core-flags/cache-flags.test.js:115]: done

File: test/core-flags/cache-flags.test.js [hunk 2]
```diff
@@ -81,6 +81,18 @@ describe('cache related flags from core', () => {
         expect(newRun.exitCode).toEqual(0);
     });
 
+    it('should assign cache build dependencies correctly when cache type is filesystem in config', () => {
+        const { stderr, stdout } = run(__dirname, ['-c', './webpack.cache.config.js']);
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain('buildDependencies: { config: [Array]');
```
Comment (evilebottnawi) [inline test/core-flags/cache-flags.test.js:87]: We need expand array here :smile: and let's add this check for other tests too

Comment (anshumanv) [reply test/core-flags/cache-flags.test.js]: done

File: packages/webpack-cli/lib/utils/flag-defaults.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        }
+        return { cache: cacheConfig };
+    }
+    return cacheConfig;
+};
+
+const assignFlagDefaults = (compilerConfig, parsedArgs) => {
+    if (Array.isArray(compilerConfig)) {
+        return compilerConfig.map((config) => cacheDefaults(config, parsedArgs));
+    } else return cacheDefaults(compilerConfig, parsedArgs);
```
Comment (jamesgeorge007) [inline packages/webpack-cli/lib/utils/flag-defaults.js:19]:
```suggestion
    if (Array.isArray(compilerConfig)) {
        return compilerConfig.map((config) => cacheDefaults(config, parsedArgs));
    }
    return cacheDefaults(compilerConfig, parsedArgs);
```

**Review by evilebottnawi — APPROVED** [review]
/cc @webpack/cli-team

**Review by snitin315 — APPROVED** [review]

**Review by evilebottnawi — APPROVED** [review]

**Pull request merged**